src/index.ts

```diff
@@ -1,228 +1,342 @@
+    validate(value) {
+      if (value.length === 0) return `Value is required!`;
+    },
+  });
 
-    folderName = folderName as string
+  if (isCancel(projectName)) {
+    onCancel('User cancelled.');
+  }
+
+  if (typeof projectName !== 'string') {
+    return `Invalid project name!`;
+  }
+
+  const placeHolderFolderName = isDojoLesson ? 'Lesson XX' : projectName;
+  const initialFolderName = isDojoLesson ? 'Lesson 0X' : projectName;
+
+  let folderName = await text({
+    message: 'In which sub-folder should this project be placed?',
+    placeholder: placeHolderFolderName,
+    initialValue: initialFolderName,
+    validate(value) {
+      if (value.length === 0) return `Value is required!`;
+    },
+  });
 
-    await tasks([
-        {
-            title: 'Setting up Git repository',
-            task: async (message) => {
-                await myExec(`git clone https://github.com/sw-craftsmanship-dojo/ns_white_crane_white_belt.git .`)
-                
-                await promises.mkdir(path.join('.', folderName));
-                await promises.mkdir(path.join('.', folderName, 'code'));
-                await promises.mkdir(path.join('.', folderName, 'theory'));
+  if (isCancel(folderName)) {
+    onCancel('User cancelled.');
+  }
 
-                process.chdir(path.join(folderName, 'code'))
+  folderName = folderName as string;
 
-                await promises.writeFile(".gitignore", gitIgnore);
+  const setupTasks = [];
 
-                return `Installed repository`;
-            },
+  if (isDojoLesson) {
+    const safeBranchName = folderName.replace(/\s+/g, '_');
+
+    setupTasks.push(
+      {
+        title: `Creating project folder '${CURRENT_DOJO}' if needed`,
+        task: async () => {
+          const dojoFolder = path.join('.', CURRENT_DOJO);
+
+          if (!existsSync(dojoFolder)) {
+            await promises.mkdir(dojoFolder);
+          } else {
+            log.info('Folder already exists, skipping creation');
+          }
+
+          process.chdir(path.join('.', CURRENT_DOJO));
         },
-        {
-            title: 'Setting up npm package',
-            task: async (message) => {
-                await myExec(`npm init -init-version=0.0.1 -y && npm install ts-node && npm install --save-dev jest jest-each ts-jest @types/jest`)
-
-                await updateValues('./package.json', new Map<string, any>([
-                    ['name', projectName],
-                    ['type', "module"],
-                    ['scripts', {test: "jest", lint: "eslint . --ext .ts,.tsx --fix"}]
-                ]));
-
-                return `Installed via npm`;
-            },
+      },
+      {
+        title: 'Clone Dojo repo or continue with existing repo',
+        task: async () => {
+          const gitFolderExists = existsSync(path.join('.', '.git'));
+
+          if (!gitFolderExists) {
+            await myExec(`git clone git@github.com:sw-craftsmanship-dojo/${CURRENT_DOJO}.git .`);
```

Comment:
This will try a git clone over ssh first. Then if it catches an error, it will try it again over HTTPS. If it still fails it will just die on a thrown error.

```suggestion
              await myExec(`git clone git@github.com:sw-craftsmanship-dojo/${CURRENT_DOJO}.git .`)
                .catch(() => {
                  // trying over https
                  log.warning("Cloning over SSH failed, trying HTTPS!");
                  myExec(`git clone https://github.com/sw-craftsmanship-dojo/${CURRENT_DOJO}.git .`);
                });
```